src/coreclr/vm/threads.cpp

```diff
@@ -4053,7 +4053,7 @@ void Thread::SetExposedObject(OBJECTREF exposed)
 {
     CONTRACTL {
         NOTHROW;
-        if (GetThreadNULLOk()) {GC_TRIGGERS;} else {DISABLED(GC_NOTRIGGER);}
+        if (GetThreadNULLOk()) {DISABLED(GC_NOTRIGGER);} else {GC_TRIGGERS;}
```

Comment:
```suggestion
        GC_NOTRIGGER;
        MODE_COOPERATIVE;
```
Also, this is only called with non-null `exposed`, so the implementation can be simplified.